### PR TITLE
Add end to end testing

### PR DESCRIPTION
**Pull request by ludwigschubi**
## Description
This PR adds:
* End2end tests for drive + login

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SKIP_PREFLIGHT_CHECK=true
```
Comment (bejow) [inline .env]: What is this for?

File: .eslintrc.json [hunk 2]
```diff
@@ -7,7 +7,11 @@
     "extends": "google",
     "globals": {
         "Atomics": "readonly",
-        "SharedArrayBuffer": "readonly"
```
Comment (bejow) [inline .eslintrc.json]: What is this for?

File: jest-puppeteer.config.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+    launch: {
+        headless: false,
```
Comment (bejow) [inline jest-puppeteer.config.js]: Should be true if executed from travis

File: jest-puppeteer.config.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        slowMo: 100,
```
Comment (bejow) [inline jest-puppeteer.config.js]: Remove slowMo attribute to speed up the tests

File: package.json [hunk 5]
```diff
@@ -3,6 +3,10 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@babel/core": "^7.7.4",
```
Comment (bejow) [inline package.json]: Aren't those packages included in create-react-app?

File: package.json [hunk 6]
```diff
@@ -93,12 +99,17 @@
     "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-prettier": "^3.1.1",
     "eslint-plugin-react": "^7.14.3",
+    "faker": "^4.1.0",
     "fetch-mock": "^7.5.1",
     "husky": "^2.7.0",
+    "jest": "^24.9.0",
```
Comment (bejow) [inline package.json]: This is definitely included in create-react-app. Could mess up our dependencies

File: src/functional_components/Drive/Drive.js [hunk 7]
```diff
@@ -39,10 +39,17 @@ const Drive = ({
     useEffect(() => {
         if (!currentPath && !loadCurrentItem && webId) {
             currentPath = webId.replace('profile/card#me', '');
-            setCurrentPath(currentPath, true);
-        } else if (currentPath && currentItem && currentItem.body) {
-            currentPath =
-                currentPath.substr(0, currentPath.lastIndexOf('/')) + '/';
+            setCurrentPath(currentPath);
+        } else if (
+            !loadCurrentItem &&
```
Comment (bejow) [inline src/functional_components/Drive/Drive.js]: The condition is kind of long and could be separated

File: src/functional_components/Drive/Drive.js [hunk 8]
```diff
@@ -39,10 +39,17 @@ const Drive = ({
...
+            currentPath &&
+            currentItem &&
+            currentItem.body
+        ) {
+            currentPath = currentPath.substr(
```
Comment (bejow) [inline src/functional_components/Drive/Drive.js]: Also this current path.substring could be a util with tests

File: src/functional_components/FileView/FileView.js [hunk 9]
```diff
@@ -30,7 +30,10 @@ export const FileView = ({
         if (fileParam) {
             if (!currentItem.body || !currentPath) {
                 let options = {};
-                if (mime.getType(currentItem.url).includes('image')) {
+                if (
+                    mime.getType(currentItem.url) &&
```
Comment (bejow) [inline src/functional_components/FileView/FileView.js]: Is this necessary? I don't know the implementation of mime. GetType

File: src/functional_components/FileView/FileView.js [hunk 10]
```diff
@@ -99,11 +102,21 @@ export const FileView = ({
         </div>
     );
 
+    console.log(currentItem.url);
```
Comment (bejow) [inline src/functional_components/FileView/FileView.js]: Remove me

File: src/functional_components/FileView/FileView.js [hunk 11]
```diff
@@ -99,11 +102,21 @@ export const FileView = ({
...
+
     return (
         <Layout
             className={styles.container}
             toolbarChildrenLeft={toolbarLeft}
             toolbarChildrenRight={!isImage && currentItem ? toolbarRight : null}
+            label={
+                currentItem.url
```
Comment (bejow) [inline src/functional_components/FileView/FileView.js]: A bit too much logic whithin the component. Can you externalize it to a util function?

File: .env (hunk 1 above)
Comment (ludwigschubi) [reply .env]: for a warning that's pretty big and annoying

File: .eslintrc.json (hunk 2 above)
Comment (ludwigschubi) [reply .eslintrc.json]: There were some eslint errors with the test syntax of 
```
describe(...
  it(...
```

Comment (ludwigschubi) [reply .eslintrc.json]: I think that's what it was but im not sure

File: jest-puppeteer.config.js (hunk 3 above)
Comment (ludwigschubi) [reply jest-puppeteer.config.js]: yeah somehow the tests fail if its headless

File: jest-puppeteer.config.js (hunk 4 above)
Comment (ludwigschubi) [reply jest-puppeteer.config.js]: good point

File: .env (hunk 1 above)
Comment (bejow) [reply .env]: About dependencies that are already installed through create react app? :laughing:

Comment (ludwigschubi) [reply .env]: nope jest-puppeteer

File: src/functional_components/FileView/FileView.js (hunk 9 above)
Comment (ludwigschubi) [reply src/functional_components/FileView/FileView.js]: if it doesn't find a type it returns undefined

File: package.json (hunk 5 above)
Comment (ludwigschubi) [reply package.json]: i don't think so because i remember that i had to install them

File: package.json (hunk 6 above)
Comment (ludwigschubi) [reply package.json]: I removed it and then the tests didn't work anymore, so i guess once you install it explicitly it regards package.json as the single source of truth

Comment (ludwigschubi) [reply package.json]: ^not true 

deleting node_modules and package.lock.json and then running `npm install` did the trick

**Comment by ludwigschubi** [comment]
Please review again @bejow

File: jest-puppeteer.config.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
+const path = require('path');
+
+module.exports = {
+    launch: {
+        headless: true,
+        slowMo: 20,
```
[the remark this answers is not in the document]
Comment (ludwigschubi) [reply jest-puppeteer.config.js]: at full speed it becomes flaky

**Comment by ludwigschubi** [comment]
@bejow please re-review. As mentioned leaving some slow-mo adds a little bit of safety if the browser is a little bit slow with loading dom nodes

**Comment by bejow** [comment]
The command `npm run test:e2e` is failing. Can you investigate?

**Comment by ludwigschubi** [comment]
@bejow tests are now being executed and stable in travis. Please review latest changes

**Pull request merged**